### PR TITLE
Adjust gatekeeper to use new VSLS policy API

### PR DESCRIPTION
**Pull request by olegoid**
Please do not merge this PR until we have the new policy API released in VSLS.

The original version of `gatekeeper` is primarily based on configuration flags. The version in this PR is modified to work with VSLS policy API, that is a re-fined version of config flags that supports a couple of extra features:

* Config merge (for situations where we might have multiple extensions competing for the same setting)
* Enforcement of domain restrictions and terminal read/write permissions

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -100,40 +23,60 @@ export async function activate(context: vscode.ExtensionContext) {
   }
 }
 
-function getAllowedDomains(): string[] {
-  // Attempt to read the list of allowed domains
-  // from the central policy file (if it exists).
-  const filePath = path.join(os.homedir(), POLICY_FILE);
-  if (fs.existsSync(filePath)) {
-    const contents = fs.readFileSync(filePath, "utf8");
+class TestPolicyProvider implements vsls.PolicyProvider {
```
Comment (lostintangent) [inline src/extension.ts:26]: Maybe `GatekeeperPolicyProvider`? We actually have customers using this, so I wouldn't want the code to somehow indicate that this is just a sample. It's actually meant for product use.

File: src/extension.ts [hunk 2]
```diff
@@ -8,90 +8,13 @@ import ActivityLog from "./activityLog";
...
-
-      // If the incoming user is from a different domain,
-      // which hasn't been whitelisted, then remove them.
-      if (!allowedDomains.includes(emailDomain)) {
-        removeUser(peer.peerNumber);
-      }
-    });
-  });
+  let policyProvider = new TestPolicyProvider();
+  api.registerPolicyProvider("Sample Provider", policyProvider);
```
Comment (lostintangent) [inline src/extension.ts:17]: Maybe `Gatekeeper Provider`?

Comment (olegoid) [reply src/extension.ts]: Works for me :)

File: src/extension.ts [hunk 3]
```diff
@@ -8,90 +8,13 @@ import ActivityLog from "./activityLog";
...
-
-      // If the incoming user is from a different domain,
-      // which hasn't been whitelisted, then remove them.
-      if (!allowedDomains.includes(emailDomain)) {
-        removeUser(peer.peerNumber);
-      }
-    });
-  });
+  let policyProvider = new GatekeeperPolicyProvider();
+  api.registerPolicyProvider("Gatekeeper Provider", policyProvider);
```
Comment (lostintangent) [inline src/extension.ts:17]: Just to confirm: this method returns a disposeable? That when disposed, would remove the old policies from Live Share? And then we could re-register the policy provider again to "refresh" the settings?

Comment (olegoid) [reply src/extension.ts]: Yep, I showcased this scenario in my latest commit.

**Pull request merged**